cognite/client/data_classes/data_modeling/query.py

```diff
@@ -205,6 +236,58 @@ def __init__(
         self.direction = direction
         self.chain_to = chain_to
         self.skip_already_deleted = skip_already_deleted
+        self.sync_mode = sync_mode
+        self.backfill_sort = backfill_sort
+
+    @staticmethod
+    def _load_sync_mode(sync_mode: str | None) -> SyncMode | None:
+        match sync_mode:
+            case None:
+                return None
+            case "onePhase":
+                return "one_phase"
+            case "twoPhase":
+                return "two_phase"
+            case "noBackfill":
+                return "no_backfill"
+            case _:
+                raise ValueError(f"Invalid sync mode {sync_mode}")
+
+    @staticmethod
+    def _dump_sync_mode(sync_mode: SyncMode, camel_case: bool = True) -> str:
+        match sync_mode:
+            case "one_phase":
+                return "onePhase" if camel_case else "one_phase"
+            case "two_phase":
+                return "twoPhase" if camel_case else "two_phase"
+            case "no_backfill":
+                return "noBackfill" if camel_case else "no_backfill"
+            case _:
+                raise ValueError(f"Invalid sync mode {sync_mode}")
```

Comment:
To have mypy ensure that the match is exhaustive
```suggestion
                assert_never(sync_mode)
```

Reply:
`assert_never` is from the `typing` module.

Reply:
Looks like `assert_never` was added in 3.11 (fails on 3.10) :disappointed:

Reply:
That backports a lot of these methods

Reply:
Trying